src/include/wiredtiger.in

```diff
@@ -4647,6 +4664,179 @@ struct __wt_file_handle {
+ * information.
+ *
+ * @snippet ex_shared_storage.c WT_SHARED_STORAGE register
+ */
+struct __wt_shared_storage {
+	/*!
+	 * Return a location handle from a location string.
+         * A location string may encode a bucket name, or the equivalent for this
+         * shared storage system, authorization information for that bucket,
+         * naming prefixes to be used for objects in that bucket, etc.
+	 *
+	 * @errors
+	 *
+	 * @param shared_storage the WT_SHARED_STORAGE
+	 * @param session the current WiredTiger session
+	 * @param location the location string
+	 * @param[out] location_handle the allocated handle
+	 */
+	int (*ss_location_handle)(WT_SHARED_STORAGE *shared_storage,
+	    WT_SESSION *session, const char *location, void **location_handle);
+
+	/*!
+	 * Free a location handle created by WT_SHARED_STORAGE::ss_location_handle.
+	 *
+	 * @errors
+	 *
+	 * @param shared_storage the WT_SHARED_STORAGE
+	 * @param session the current WiredTiger session
+	 * @param location_handle the handle to be freed.
+	 */
+	int (*ss_location_handle_free)(WT_SHARED_STORAGE *shared_storage,
+	    WT_SESSION *session, void *location_handle);
+
+	/*!
+	 * Return a list of object names for the given location.
+	 *
+	 * @errors
+	 *
+	 * @param shared_storage the WT_SHARED_STORAGE
+	 * @param session the current WiredTiger session
+	 * @param location_handle the location to list
+	 * @param[out] objlist the method returns an allocated array of
+	 *    individually allocated strings, one for each object in the location.
+	 * @param[out] countp the number of entries returned
+	 */
+	int (*ss_location_list)(WT_SHARED_STORAGE *shared_storage,
+	    WT_SESSION *session, void *location_handle,
+	    char ***objlist, uint32_t *countp);
+
+	/*!
+	 * Free memory allocated by WT_SHARED_STORAGE::location_list.
+	 *
+	 * @errors
+	 *
+	 * @param shared_storage the WT_SHARED_STORAGE
+	 * @param session the current WiredTiger session
+	 * @param objlist array returned by WT_SHARED_STORAGE::location_list
+	 * @param count count returned by WT_SHARED_STORAGE::location_list
+	 */
+	int (*ss_location_list_free)(WT_SHARED_STORAGE *shared_storage,
+	    WT_SESSION *session, char **objlist, uint32_t count);
+
+	/*!
+	 * Return if the named object exists in the location.
+	 *
+	 * @errors
+	 *
+	 * @param shared_storage the WT_SHARED_STORAGE
+	 * @param session the current WiredTiger session
+         * @param location_handle the location to search
+	 * @param name the name of the object
+	 * @param[out] existp If the named shared storage object exists
+	 */
+	int (*ss_exist)(WT_SHARED_STORAGE *shared_storage,
+	    WT_SESSION *session, void *location_handle, const char *name, bool *existp);
+
+	/*!
+	 * Open a handle for a named shared storage object
+	 *
+	 * The method should return ENOENT if the object is not being created and
+	 * does not exist.
+	 *
+	 * The method should return EACCES if the object cannot be opened given
+         * permissions by the location.
```

Comment:
Do you know why the strange indentation is going on (in several locations)?

Reply:
Fixed. The indentations are tabs vs spaces.  Apparently, clang_format doesn't run on wiredtiger.in .